internal/runbits/runtime/requirements/install.go

```diff
@@ -0,0 +1,448 @@
+				"Note: the actual package name ({{.V0}}) is different from the requested package name ({{.V1}})",
+				name, req.Name))
+		}
+		jsonObjects[i] = &struct {
+			Name      string `json:"name"`
+			Version   string `json:"version,omitempty"`
+			Type      string `json:"type"`
+			Operation string `json:"operation"`
+		}{
+			name,
+			req.Version,
+			nsType,
+			req.Operation.String(),
+		}
+	}
+	r.Output.Print(output.Prepare(strings.Join(messages, "\n"), jsonObjects))
+
+	r.Output.Notice("") // blank line
+	r.Output.Notice(locale.T("operation_success_local"))
+
+	return nil
+}
+
+func (r *RequirementOperation) searchForRequirements(requirements []*Requirement) (results []*model.IngredientAndVersion, rerr error) {
+	results = make([]*model.IngredientAndVersion, len(requirements))
+
+	names := make([]string, len(requirements))
+	for i, req := range requirements {
+		names[i] = req.Name
+	}
+
+	pg := output.StartSpinner(r.Output, locale.Tr("progress_search", strings.Join(names, ", ")), constants.TerminalAnimationInterval)
+	defer func() {
+		if rerr != nil {
+			pg.Stop(locale.T("progress_fail"))
+		}
+	}()
+
+	for i, req := range requirements {
+		ingredients, err := r.searchForIngredient(req, req.Namespace)
+		if err != nil {
+			return nil, errs.Wrap(err, "Unable to search for requirement '%s'", req.Name)
+		}
+
+		// Collect all ingredients into a unique list of requirement names.
+		unique := make([]*model.IngredientAndVersion, 0)
+		for _, i := range ingredients {
+			if !funk.Contains(unique, i) {
+				unique = append(unique, i)
+			}
+		}
+
+		if len(unique) == 0 {
+			suggestions, err := getSuggestions(req.Namespace, req.Name, r.Auth)
+			if err != nil {
+				multilog.Error("Failed to retrieve suggestions: %v", err)
+			}
+
+			if len(suggestions) == 0 {
+				return nil, &ErrNoMatches{
+					locale.WrapExternalError(err, "package_ingredient_alternatives_nosuggest", "", req.Name),
+					req.Name, nil}
+			}
+
+			return nil, &ErrNoMatches{
+				locale.WrapExternalError(err, "package_ingredient_alternatives", "", req.Name, strings.Join(suggestions, "\n")),
+				req.Name, ptr.To(strings.Join(suggestions, "\n"))}
+		}
+
+		results[i] = unique[0]
+	}
+
+	pg.Stop(locale.T("progress_found"))
+	return results, nil
+}
+
+func (r *RequirementOperation) searchForIngredient(req *Requirement, namespace *model.Namespace) ([]*model.IngredientAndVersion, error) {
+	name := req.Name
+	ns := ""
+	caseSensitive := false
+
+	if namespace != nil {
+		if normalized, err := model.FetchNormalizedName(*namespace, name, r.Auth); err == nil {
+			name = normalized
+		} else {
+			multilog.Error("Failed to normalize '%s': %v", req.Name, err)
+		}
+		ns = namespace.String()
+		//caseSensitive = true // ideally case-sensitive would be true (PB-4371)
+	}
+
+	return model.SearchIngredientsStrict(ns, name, caseSensitive, false, nil, r.Auth)
+}
+
+func (r *RequirementOperation) InstallPlatform(requirements []*Requirement) (rerr error) {
```

Comment:
For readability this needs to be a separate function because platforms are not requirements, so there are completely different code paths.